addons/rating/models/rating.py

```diff
@@ -57,7 +68,16 @@ def _compute_rating_image(self):
     def _compute_rating_text(self):
         text = {10: _('Satisfied'), 5: _('Not satisfied'), 1: _('Highly dissatisfied')}
         for rating in self:
-            rating.rating_text = text[rating.rating] or _('No rating yet')
+            rating.rating_text = rating.rating and text[rating.rating] or _('No rating yet')
+
+    @api.model
+    def create(self, vals):
+        record = super(Rating, self).create(vals)
+        res = self.env[record.res_model].browse(record.res_id)
+        record.sudo().write(res.get_parent_res_id_value())
+        parent_res_model_name = res.get_parent_model_name(vals)
+        record.parent_res_model_id = self.env['ir.model'].search([('model', '=', parent_res_model_name)]).id
```

Comment:
Rather than make a `write` rigth after the `create`, shouldn't it be better to update the values given to the create directly ? Like it is done in mail_alias.py